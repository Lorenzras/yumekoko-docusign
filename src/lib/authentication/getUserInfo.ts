import {apiClient} from '../../config';
import {getAccessToken} from './getAccessToken';

/**
 * Get user info.
 *
 * @returns {IGetUserInfoResponse} returns Userinfo
 */
export const getUserInfo = async () => {
  const {accessToken} = await getAccessToken();
  const userInfo: IGetUserInfoResponse = await apiClient
    .getUserInfo(accessToken);


  return userInfo;
};
