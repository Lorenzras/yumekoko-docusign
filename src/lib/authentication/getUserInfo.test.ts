import {getUserInfo} from './getUserInfo';

describe('Base URI', ()=>{
  it('should get base uri', async ()=>{
    const response = await getUserInfo();
    console.log('base uri', response.accounts[0].baseUri);
    expect(response).toMatchSnapshot();
  });
});
