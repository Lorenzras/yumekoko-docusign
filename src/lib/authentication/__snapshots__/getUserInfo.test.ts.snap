// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base URI should get base uri 1`] = `
exports {
  "accounts": Array [
    exports {
      "accountId": "d1872cc7-b51b-45fc-b401-498f7cb641ed",
      "accountName": "Individual",
      "baseUri": "https://demo.docusign.net",
      "isDefault": "true",
      "organization": exports {
        "links": Array [
          exports {
            "href": "https://account-d.docusign.com/organizations/020a334d-43a7-4d92-9879-9e8af32ce1bb",
            "rel": "self",
          },
        ],
        "organization_id": "020a334d-43a7-4d92-9879-9e8af32ce1bb",
      },
    },
  ],
  "created": "2022-06-24T06:37:38.24",
  "email": "lenzras@gmail.com",
  "familyName": "Ras",
  "givenName": "Lorenz",
  "name": "Lorenz Ras",
  "sub": "9169dfa2-cb51-47e7-bf93-da59d152da90",
}
`;
